invenio_rdm_records/services/generators.py

```diff
@@ -86,6 +116,44 @@ def query_filter(self, **kwargs):
             return q_public
 
 
+class IfDraft(ConditionalGenerator):
```

Comment:
I've asked in rdm-internal and nobody seems to be using this, plus this is the only usage in the code base. History tells that it was a temporary generator, replaced later by splitting the actions into `can_..._files` (for records) and `can_draft_..._files).

Should we remove it?

Reply:
maybe deprecating it would be safer, especially because it goes as far as the latest LTS?

Reply:
I can add a deprecation warning. But IMO, we can just merge as is and do the deprecation in another PR if needed since it is not the actual goal of it, but more of an aftermath.